src/components/autosuggest/Autosuggest.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, FC } from "react";
+import sn from "classnames";
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownMenu,
+  DropdownItem,
+  Input
+} from "reactstrap";
+
+import "./Autosuggest.scss";
+
+interface Item {
+  label: string;
+  value: string;
+  id: number | string;
+}
+
+interface Autosuggest {
+  collection?: Array<Item>;
+  placeholder?: string;
+  searchPlaceholder?: string;
+  value: string;
+  onChangeValue: (value: string) => void;
+}
+
+const blockName = "autosuggest";
+
+export const Autosuggest: FC<Autosuggest> = ({
+  collection,
+  placeholder,
+  searchPlaceholder,
+  value,
+  onChangeValue
+}) => {
+  const [isOpen, toggleOpen] = useState(false);
+  const [list, setCollection] = useState(collection);
+
+  const toggleDropdown = () => toggleOpen(!isOpen);
+
+  const onSelect = (e: any) => {
```

Comment:
to avoid unnecessary renders it is better to wrap callbacks with useCallback hook